src/formatter-settings/index.ts

```diff
@@ -4,27 +4,347 @@
+                    throw new Error(`Invalid setting value in ${setting.id}`);
+                }
+                setting.value = webViewValue;
+            }
+            webviewPanel.webview.postMessage({
+                command: "loadSetting",
+                setting: setting,
+            });
+        }
+    }
+
+    private async inheritVSCodeSettings(webviewPanel: vscode.WebviewPanel) {
+        const javaTabSize: number | undefined = await this.findJavaSpecificSetting<number>("editor.tabSize");
+        const javaSpacePolicy: boolean | undefined = await this.findJavaSpecificSetting<boolean>("editor.insertSpaces");
+        const vscodeTabSize: number | undefined = vscode.workspace.getConfiguration().get<number>("editor.tabSize");
+        const vscodeSpacePolicy: boolean | undefined = vscode.workspace.getConfiguration().get<boolean>("editor.insertSpaces");
+        const tabSize = (javaTabSize === undefined) ? vscodeTabSize : javaTabSize;
+        const spacePolicy = (javaSpacePolicy === undefined) ? vscodeSpacePolicy : javaSpacePolicy;
+        for (const setting of initializeSupportedVSCodeSettings()) {
+            if (setting.id === SupportedSettings.TABULATION_CHAR) {
+                setting.value = (spacePolicy === false) ? "tab" : "space";
+                this.profileSettings.set(setting.id, setting.value);
+            } else if (setting.id === SupportedSettings.TABULATION_SIZE) {
+                setting.value = String(tabSize);
+                this.profileSettings.set(setting.id, setting.value);
+            }
+            webviewPanel.webview.postMessage({
+                command: "loadSetting",
+                setting: setting,
+            });
+        }
+    }
+
+    private async findJavaSpecificSetting<T>(setting: string): Promise<T | undefined> {
+        let result: T | undefined;
+        const config = vscode.workspace.getConfiguration().get<Object>("[java]");
+        if (!config) {
+            return result;
+        }
+        const entries = Object.entries(config);
+        for (const entry of entries) {
+            if (entry[0] === setting) {
+                result = entry[1];
+            }
+        }
+        return result;
+    }
+
+    private async formatWithProfileSettings(webviewPanel: vscode.WebviewPanel) {
+        const content = await vscode.commands.executeCommand<string>("java.execute.workspaceCommand", "java.edit.stringFormatting", Example.getExample(this.exampleKind), JSON.stringify([...this.profileSettings]), this.settingsVersion);
+        webviewPanel.webview.postMessage({
+            command: "formattedContent",
+            content: content,
+        });
+    }
+
+    private async changeProfileSetting(id: string, value: string, document: vscode.TextDocument): Promise<void> {
+        const profileElement = this.profileElements.get(id);
+        if (!profileElement) {
+            // add a new setting not exist in the profile
+            if (!this.profileSettingsArray.length) {
```

Comment:
What will happen if the inserted setting in not supported in the current version of the profile

Reply:
Will add a check for the inserted setting, if not supported in current version, will throw an error message.